tests/binaries/makefile

```diff
@@ -0,0 +1,36 @@
+CC              =       gcc
+DEBUG           =       1
+CFLAGS         +=       -Wall
+SOURCES         =       $(wildcard *.c)
+COMPILED        =       $(SOURCES:.c=.o)
+LINKED          =       $(SOURCES:.c=.out)
+LDFLAGS         =
+EXTRA_FLAGS	    =
+
+ifeq ($(TARGET), x86)
+CFLAGS          +=      -m32
+endif
+
+ifeq ($(DEBUG), 1)
+CFLAGS          +=      -DDEBUG=1 -ggdb -O0
+else
+CFLAGS          +=      -O1
+endif
+
+
+.PHONY : all clean
+
+all: $(LINKED)
+
+
+%.out : %.c
+	@echo "[+] Building '$@'"
+	@$(CC) $(CFLAGS) $(EXTRA_FLAGS) -o $@ $? $(LDFLAGS)
+
+clean :
+	@echo "[+] Cleaning stuff"
+	@rm -f $(COMPILED) $(LINKED)
+
+
+reference-binary.out: EXTRA_FLAGS := -Dexample=1
```

Comment:
the `-D` here is not needed but I wanted to leave it here just to show up that extra flags can be filled